diff -up solid-6.28.0/src/solid/devices/frontend/devicemanager.cpp.1~ solid-6.28.0/src/solid/devices/frontend/devicemanager.cpp
--- solid-6.28.0/src/solid/devices/frontend/devicemanager.cpp.1~
+++ solid-6.28.0/src/solid/devices/frontend/devicemanager.cpp
@@ -88,8 +88,23 @@ QList<Solid::Device> Solid::Device::listFromType(const DeviceInterface::Type &type, const QString &parentUdi)
         }
 
         const auto udis = backend->devicesFromQuery(parentUdi, type);
-        for (const auto &udi : udis) {
-            list.append(Device(udi));
+        if (type == DeviceInterface::Type::Processor) {
+            QMap<int, QString> coresMap;
+            QRegularExpression re(QStringLiteral("cpu([0-9]+)"));
+            for (const auto &udi : udis) {
+                QRegularExpressionMatch match = re.match(udi);
+                if (match.hasMatch()) {
+                    QString matched = match.captured(1);
+                    coresMap[matched.toUInt()] = udi;
+                }
+            }
+            for (auto ii = coresMap.constBegin(); ii != coresMap.constEnd(); ++ii) {
+                list.append(Device(ii.value()));
+            }
+        } else {
+            for (const auto &udi : udis) {
+                list.append(Device(udi));
+            }
         }
     }
 
diff -up solid-6.28.0/src/solid/devices/frontend/devicemanager_p.h.1~ solid-6.28.0/src/solid/devices/frontend/devicemanager_p.h
--- solid-6.28.0/src/solid/devices/frontend/devicemanager_p.h.1~
+++ solid-6.28.0/src/solid/devices/frontend/devicemanager_p.h
@@ -13,6 +13,7 @@
 
 #include <QHash>
 #include <QPointer>
+#include <QRegularExpression>
 #include <QSharedData>
 #include <QThreadStorage>
 
